apps/admin/src/pages/sponsors/update-sponsor-form.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState, useEffect } from "react";
+import type { SubmitHandler } from "react-hook-form";
+import type { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import type { RouterOutput } from "@knighthacks/api";
+import { insertSponsorSchema } from "@knighthacks/db";
+import { Button } from "~/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "~/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "~/components/ui/select";
+import { Input } from "~/components/ui/input";
+import { trpc } from "~/trpc";
+
+type Sponsor = RouterOutput["sponsors"]["getAll"][number];
+
+const updateSponsorFormSchema = insertSponsorSchema;
+
+type updateSponsorFormValues = z.infer<typeof updateSponsorFormSchema>
+
+export function UpdateSponsorForm({sponsor }: {sponsor: Sponsor}) {
+  const [hackathonIds, setHackathonIds] = useState<number[]>([]);
+  const utils = trpc.useUtils();
+
+  const form = useForm<updateSponsorFormValues>({
+    resolver: zodResolver(updateSponsorFormSchema),
+    defaultValues: {
+      name: sponsor.name,
+      website: sponsor.website,
+      logo: sponsor.logo,
+      tier: sponsor.tier,
+      hackathonId: sponsor.hackathonId
+    },
+  });
+
+  const { mutate, isLoading } = trpc.sponsors.update.useMutation({
+    onSuccess: async () => {
+      await utils.sponsors.getAll.invalidate();
+      toast("Success!", {
+        description: "User updated",
+      });
+    },
+    onError: (error) => {
+      toast("Error!", {
+        description: error.message,
+      });
+    },
+  });
+
+  const onSubmit: SubmitHandler<updateSponsorFormValues> = (values) => {
+    mutate({ id: sponsor.id, ...values })
+  };
+
+  // fill array with all valid hackathonIds
+  const { data: hackathons } = trpc.hackathons.getAll.useQuery();
+
+  // useEffect to call getHackathonIds on component mount
+  useEffect(() => {
```

Comment:
I dont think this useEffect is necessary since you just iterate through each hackathon and set the value of the select item by accessing the id property of each hackathon.

Reply:
Whenever I dont use the useEffect I get infinite renders